lms/djangoapps/student_account/views.py

```diff
@@ -302,6 +306,51 @@ def _external_auth_intercept(request, mode):
         return external_auth_register(request)
 
 
+def get_user_orders(user):
+    """Given a user, get the detail of all the orders from the Ecommerce service.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting ecommerce.
+
+    Returns:
+        list of dict, representing orders returned by the Ecommerce service.
+    """
+    no_data = []
+    user_orders = []
+    allowed_course_modes = ['professional', 'verified', 'credit']
+    commerce_configuration = CommerceConfiguration.current()
+    # user_query = {'username': user.username}
+
+    use_cache = commerce_configuration.is_cache_enabled
+    cache_key = commerce_configuration.CACHE_KEY + '.' + str(user.id) if use_cache else None
+    api_client = ecommerce_api_client(user)
+    commerce_user_orders = get_edx_api_data(
+        commerce_configuration, api_client, 'orders', querystring=None, cache_key=cache_key
+    )
+
+    for order in commerce_user_orders:
+        if order['status'].lower() == 'complete':
+            for attribute in order['lines'][0]['product']['attribute_values']:
+                if attribute['name'] == 'certificate_type' and attribute['value'] in allowed_course_modes:
+                    try:
+                        order_data = {
+                            'number': order['number'],
+                            'price': order['total_excl_tax'],
+                            'title': order['lines'][0]['title'],
+                            'order_date': ModuleI18nService().strftime(
```

Comment:
Because we want to use edx default `SHORT_DATE` format which is implemented [here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/date_utils.py#L104) and being used by [this method](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/django.py#L296) in [this class](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/django.py#L255)